### PR TITLE
fix(cli): Interval validation for `meltano schedule set`

### LINKED ISSUE #7814: bug: `meltano schedule set` does not validate interval
### Meltano Version

2.19.1

### Python Version

3.10

### Bug scope

CLI (options, error messages, logging, etc.)

### Operating System

N/A

### Description

If you pass an invalid interval to `meltano schedule add`, it will error:
```bash
(meltano-dev-3.10) braedon ~/meltano/projects/test-project $ meltano schedule add csv-to-jsonl --interval fakeinterval --extractor tap-csv --loader target-jsonl
Invalid Cron expression or alias: 'fakeinterval'. Please use a valid cron expression.
```

The same is not true for `meltano schedule set`:
```bash
(meltano-dev-3.10) braedon ~/meltano/projects/test-project $ meltano schedule set csv-to-jsonl --interval fakeinterval
Updated schedule 'csv-to-jsonl'
(meltano-dev-3.10) braedon ~/meltano/projects/test-project $ meltano schedule list
[fakeinterval] elt csv-to-jsonl: tap-csv → target-jsonl x transforms
```

The same validation should be done when setting the interval with `set` and `add`.

### Code

_No response_

### PR DESCRIPTION
**Pull request by XshubhamX**
1. Used same validations that we use for `meltano schedule add`
https://github.com/meltano/meltano/blob/main/src/meltano/core/schedule_service.py#L216

cc @BraedonLeonard @edgarrmondragon 

Closes #7814

### CODE REVIEW

**Comment by BraedonLeonard** [comment]
Looks good to me! You'll need to add a unit test to ensure that validation is working as intended, and once you do I'll take another look and approve it 👍

File: src/meltano/cli/schedule.py
Comment (edgarrmondragon) [inline src/meltano/cli/schedule.py:294]: You can probably catch an invalid interval earlier by using a [`click.ParamType`](https://click.palletsprojects.com/en/8.1.x/api/#click.ParamType) for the `--interval` option. Wdyt?

Comment (XshubhamX) [reply src/meltano/cli/schedule.py]: It works @edgarrmondragon , nice suggestion @ @

**Comment by XshubhamX** [comment]
@edgarrmondragon @BraedonLeonard updated the PR

File: src/meltano/cli/schedule.py
Comment (edgarrmondragon) [inline src/meltano/cli/schedule.py:368]: I think this code is unreachable, i.e. if `--interval` is not passed to the command, this method will not even be called and if `--interval` is indeed passed, then `value` is not `None`.

It can probably be removed:

```suggestion
```

**Review by BraedonLeonard — COMMENTED** [review]
Look good to me 👍 I'll leave the final approval up to @edgarrmondragon

File: src/meltano/cli/schedule.py
Comment (XshubhamX) [reply src/meltano/cli/schedule.py:368]: Make sense

**Review by edgarrmondragon — APPROVED** [review]

**Review by edgarrmondragon — APPROVED** [review]
Thanks @XshubhamX!

**Pull request merged**
